cmd/podman/create_cli.go

```diff
@@ -74,6 +75,77 @@ func addWarning(warnings []string, msg string) []string {
 	return append(warnings, msg)
 }
 
+// Format supported.
+// podman run --mount type=bind,src=/etc/resolv.conf,target=/etc/resolv.conf ...
+// podman run --mount type=tmpfs,target=/dev/shm ..
+func parseMounts(mounts []string) ([]spec.Mount, error) {
+	var mountList []spec.Mount
+	errInvalidSyntax := errors.Errorf("incorrect mount format : should be --mount type=<bind|tmpfs>,src=<host-dir>,target=<ctr-dir>,[options]")
+	for _, mount := range mounts {
+		var tokenCount int
+		var mountInfo spec.Mount
+
+		arr := strings.SplitN(mount, ",", 2)
+		if len(arr) < 2 {
+			return nil, errInvalidSyntax
+		}
+		kv := strings.Split(arr[0], "=")
+		if err := validateFSType(kv[1]); err != nil {
+			return nil, err
+		}
+
+		if kv[1] == "bind" {
+			mountInfo.Type = string(cc.TypeBind)
+			tokenCount = 3
+		} else if kv[1] == "tmpfs" {
+			mountInfo.Type = string(cc.TypeTmpfs)
+			tokenCount = 2
+		}
+
+		tokens := strings.SplitN(arr[1], ",", tokenCount)
+		if len(tokens) < (tokenCount - 1) {
+			return nil, errInvalidSyntax
+		}
+		for i, val := range tokens {
+			if i == (tokenCount - 1) {
+				//Parse tokens before options.
+				break
+			}
+			kv := strings.Split(val, "=")
```

Comment:
nit and admittedly a personal one.  I'm not a big fan of reusing the same variable name like this.